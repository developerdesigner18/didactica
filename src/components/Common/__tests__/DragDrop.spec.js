import { createLocalVue, mount } from "@vue/test-utils";
import { waitRAF } from "@/../tests/utils.js";

import LxpDragDrop from "../DragDrop.vue";
import LxpDropdown from "../Dropdown.vue";
import { BDropdown, BDropdownItem, BImg } from "bootstrap-vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();

localVue.component("LxpDragDrop", LxpDragDrop);
localVue.component("b-img", BImg);
localVue.component("lxp-dropdown", LxpDropdown);
localVue.component("b-dropdown", BDropdown);
localVue.component("b-dropdown-item", BDropdownItem);

describe("DragDrop.vue", () => {
  it("render when created ", async () => {
    let wrapper = await getWrapper();
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });
  describe("Props", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    it("should add error class when `showError` is true", async () => {
      wrapper = await getWrapper();
      await wrapper.setData({ isDragging: false, hasError: true });
      await waitRAF();

      const errorDragDrop = wrapper.find(".lxp-dd__error");
      expect(errorDragDrop.exists()).toBe(true);
    });
  });
});

function getWrapper() {
  return mount(LxpDragDrop, {
    propsData: {
      noteText: "Recomended file size 1MB",
      id: "dragdrop-1"
    },
    localVue
  });
}
