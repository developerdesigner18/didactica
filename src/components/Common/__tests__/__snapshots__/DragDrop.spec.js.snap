// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragDrop.vue render when created  1`] = `
<div
  class="lxp-drag-drop"
  id="dragdrop-1"
>
  <div
    class="lxp-dd__drop variant1"
  >
    <input
      accept=".pdf,.jpg,.jpeg,.png"
      class="hidden-input"
      id="lxp-file-input-dragdrop-1"
      type="file"
    />
     
    <div
      class="h-100 w-100"
    >
      <div
        class="h-100"
      >
         
        <div
          class="lxp-dd__drop--default"
        >
          <img
            alt="upload"
            loading="lazy"
            src="https://files.lxp.academy.who.int/didactica/assets/images/drag-drop/upload.svg"
          />
           
          <div
            class="lxp-dd__drop--default--content"
          >
            
            Drag and drop image here or
            
            <label
              class="lxp-dd__drop--default-select"
              for="lxp-file-input-dragdrop-1"
            >
              Select image
            </label>
          </div>
        </div>
      </div>
       
      <!---->
       
      <!---->
    </div>
  </div>
   
  <span
    class="lxp-dd__note"
  >
    
    Recomended file size 1MB
  
  </span>
</div>
`;
